packages/fiori-elements-opa-writer/src/index.ts

```diff
@@ -3,12 +3,45 @@ import { create as createStorage } from 'mem-fs';
 import type { Editor } from 'mem-fs-editor';
 import { create } from 'mem-fs-editor';
 import type { ManifestNamespace } from '@sap-ux/ui5-config';
-import type { FEV4OPAConfig } from './types';
+import type { FEV4OPAConfig, FEV4OPAPageConfig } from './types';
+import { SupportedPageTypes } from './types';
 
 type Manifest = ManifestNamespace.SAPJSONSchemaForWebApplicationManifestFile & { [key: string]: any };
 
 /**
- * Create the configuration object from the app descriptor and the FLP app name.
+ * Create the page configuration object from the app descriptor and the target key.
+ * @param manifest - the app descriptor of the app
+ * @param targetKey - the key of the target in the manifest
+ * @returns Page configuration object, or undefined if the target type is not supported
+ */
+function createPageConfig(manifest: Manifest, targetKey: string): FEV4OPAPageConfig | undefined {
+    const appTargets = manifest['sap.ui5']?.routing?.targets;
+    const target = appTargets && appTargets[targetKey];
+    const appID = manifest['sap.app'].id;
+    const appPath = appID.split('.').join('/');
+
+    if (
+        target &&
```

Comment:
small thing: you could start with `target?.id` and then remove line 27

Reply:
done